File: api-standalone/dev-server.py
#!/usr/bin/env python3
"""
Development Server - Local Mock API for Frontend Testing
Simula o comportamento da API do ESP32 rodando localmente
"""

from flask import Flask, send_from_directory, jsonify
from flask_cors import CORS
import math
import time
import json
import os
from functools import wraps

app = Flask(__name__, static_folder='data')
CORS(app)  # Enable CORS for local development

# Arquivo para salvar credenciais
CREDENTIALS_FILE = os.path.join(os.path.dirname(__file__), 'credentials.json')

# Configuração de autenticação (valores padrão)
API_USER = 'admin'
API_PASS = 'admin123'

def load_credentials():
    """Carregar credenciais salvas do arquivo JSON"""
    global API_USER, API_PASS
    try:
        if os.path.exists(CREDENTIALS_FILE):
            with open(CREDENTIALS_FILE, 'r') as f:
                data = json.load(f)
                API_USER = data.get('username', 'admin')
                API_PASS = data.get('password', 'admin123')
                print('✓ Credenciais carregadas do arquivo credentials.json')
                print(f'   Usuário: {API_USER}')
                print('   Senha: ***')
    except Exception as e:
        print(f'⚠️ Erro ao carregar credenciais, usando valores padrão: {e}')

def save_credentials(username, password):
    """Salvar credenciais em arquivo JSON"""
    try:
        data = {
            'username': username,
            'password': password,
            'updated_at': time.strftime('%Y-%m-%d %H:%M:%S')
        }
        with open(CREDENTIALS_FILE, 'w') as f:
            json.dump(data, f, indent=2)
        print('✓ Credenciais salvas em credentials.json')
        return True
    except Exception as e:
        print(f'❌ Erro ao salvar credenciais: {e}')
        return False

# Carregar credenciais na inicialização
load_credentials()

def check_auth(username, password):
    """Verifica credenciais"""
    return username == API_USER and password == API_PASS

def authenticate():
    """Retorna resposta 401 com desafio de autenticação"""
    return jsonify({'error': 'Authentication required'}), 401, {
        'WWW-Authenticate': 'Basic realm="Login Required"'
    }

def requires_auth(f):
    """Decorator para exigir autenticação HTTP Basic"""
    @wraps(f)
    def decorated(*args, **kwargs):
        from flask import request
        auth = request.authorization
        if not auth or not check_auth(auth.username, auth.password):
            return authenticate()
        return f(*args, **kwargs)
    return decorated

def generate_demo_data():
    """Gera dados simulados realistas (mesma lógica do ESP32)"""
    current_time = time.time()
    variance = math.sin(current_time / 30.0) * 0.1  # Variação lenta
    
    # Solar PV - simulando sol parcial
    pv_voltage = 85.0 + variance * 10.0
    pv_current = 4.5 + variance * 2.0
    pv_power = pv_voltage * pv_current
    
    # Carregador
    charger_voltage = 54.2 + variance * 0.5
    charger_current = 8.2 + variance * 1.0
    charger_power = 445.0 + variance * 50.0
    
    # Bateria - simulando estado de carga
    battery_voltage = 52.8 + variance * 0.3
    battery_current = 8.0 + variance * 1.5
    battery_power = 422.0 + variance * 40.0
    battery_soc = 65.0 + variance * 5.0
    battery_temp = 25.0 + variance * 2.0
    
    # Inversor - simulando modo off-grid com carga
    ac_voltage = 220.0 + variance * 2.0
    ac_current = 2.5 + variance * 0.5
    ac_frequency = 50.0 + variance * 0.1
    ac_power = 550.0 + variance * 50.0
    load_percent = 27.5 + variance * 5.0
    dc_voltage = 52.8 + variance * 0.3
    
    # Acumuladores - incrementando lentamente
    time_hours = current_time / 3600.0
    charger_accumulated = 12.5 + (time_hours * 0.1) % 5.0
    total_charged = 245.8 + (time_hours * 0.2) % 10.0
    total_discharged = 198.3 + (time_hours * 0.15) % 8.0
    
    device_temp = 42.0 + variance * 3.0
    
    return {
        "charger": {
            "voltage": round(charger_voltage, 1),
            "current": round(charger_current, 2),
            "power": round(charger_power, 0),
            "accumulated_power": round(charger_accumulated, 1)
        },
        "pv": {
            "voltage": round(pv_voltage, 1),
            "current": round(pv_current, 2),
            "power": round(pv_power, 0)
        },
        "battery": {
            "voltage": round(battery_voltage, 1),
            "current": round(battery_current, 2),
            "power": round(battery_power, 0),
            "soc": round(battery_soc, 0),
            "temperature": round(battery_temp, 1)
        },
        "inverter": {
            "mode": "Off-Grid",
            "mode_id": 3,
            "ac_voltage": round(ac_voltage, 1),
            "ac_current": round(ac_current, 2),
            "ac_frequency": round(ac_frequency, 2),
            "ac_power": round(ac_power, 0),
            "load_percent": round(load_percent, 0),
            "dc_voltage": round(dc_voltage, 1),
            "max_charge_current": 60,
            "max_discharge_current": 60,
            "accumulated_power": round(total_discharged, 1)
        },
        "totals": {
            "total_charged": round(total_charged, 1),
            "total_discharged": round(total_discharged, 1),
            "device_temperature": round(device_temp, 1)
        },
        "last_update": int(current_time),
        "uptime": int(current_time - app.config['START_TIME']),
        "modbus_error": False,
        "demo_mode": True
    }

# ==================== ROTAS DA API ====================

@app.route('/api/sensors')
@requires_auth
def api_sensors():
    """Endpoint principal de sensores (requer autenticação)"""
    data = generate_demo_data()
    return jsonify(data)

@app.route('/api/status')
@requires_auth
def api_status():
    """Endpoint de status do dispositivo"""
    return jsonify({
        "device_name": "MUST Inverter API (Dev Server)",
        "ip_address": "127.0.0.1",
        "mac_address": "00:00:00:00:00:00",
        "wifi_ssid": "Development",
        "wifi_rssi": -42,
        "uptime_seconds": int(time.time() - app.config['START_TIME']),
        "free_heap": 200000,
        "modbus_connected": False
    })

@app.route('/api/reset', methods=['POST'])
@requires_auth
def api_reset():
    """Endpoint de reset (não faz nada no dev server)"""
    return jsonify({"message": "Reset não disponível no dev server"}), 200

@app.route('/api/credentials', methods=['POST'])
@requires_auth
def api_credentials():
    """Endpoint para alteração de credenciais"""
    from flask import request
    data = request.get_json()
    
    username = data.get('username')
    password = data.get('password')
    current_password = data.get('current_password')
    
    # Validar senha atual
    if current_password != API_PASS:
        return jsonify({'error': 'Current password is incorrect'}), 401
    
    # Validar se há algo para alterar
    if not username and not password:
        return jsonify({'error': 'No changes provided'}), 400
    
    # Validar tamanho da senha
    if password and len(password) < 6:
        return jsonify({'error': 'Password must be at least 6 characters'}), 400
    
    print('⚠️ Tentativa de alteração de credenciais:')
    print(f'   Novo usuário: {username or "(não alterado)"}')
    print(f'   Nova senha: {"***" if password else "(não alterada)"}')
    
    # Definir valores finais
    global API_USER, API_PASS
    final_username = username or API_USER
    final_password = password or API_PASS
    
    # Salvar em arquivo
    if save_credentials(final_username, final_password):
        # Atualizar variáveis globais
        API_USER = final_username
        API_PASS = final_password
        
        return jsonify({
            'success': True,
            'message': 'Credentials updated successfully',
            'username': final_username,
            'note': 'Credenciais salvas em credentials.json'
        }), 200
    else:
        return jsonify({
            'error': 'Failed to save credentials'
        }), 500

# ==================== SERVIR ARQUIVOS ESTÁTICOS ====================

@app.route('/')
@requires_auth
def index():
    """Serve o dashboard principal (requer autenticação)"""
    return send_from_directory('data', 'index.html')

@app.route('/index.html')
@requires_auth
def index_html():
    """Serve index.html com autenticação"""
    return send_from_directory('data', 'index.html')

@app.route('/config.html')
def config():
    """Serve a página de configuração (modo AP - sem auth)"""
    return send_from_directory('data', 'config.html')

@app.route('/settings.html')
@requires_auth
def settings():
    """Serve a página de configurações (requer autenticação)"""
    return send_from_directory('data', 'settings.html')

@app.route('/css/<path:filename>')
def serve_css(filename):
    """Serve arquivos CSS"""
    return send_from_directory('data/css', filename)

@app.route('/js/<path:filename>')
def serve_js(filename):
    """Serve arquivos JavaScript"""
    return send_from_directory('data/js', filename)

# ==================== INICIALIZAÇÃO ====================

if __name__ == '__main__':
    app.config['START_TIME'] = time.time()
    
    print("\n" + "="*60)
    print("  🚀 MUST Inverter API - Development Server")
    print("="*60)
    print("\n📌 Servidor rodando em modo de desenvolvimento")
    print("   URL: http://localhost:5000")
    print("\n🔐 Credenciais de autenticação:")
    print(f"   Usuário: {API_USER}")
    print(f"   Senha: {API_PASS}")
    print("\n📁 Servindo arquivos de: ./data/")
    print("   • HTML: index.html, config.html")
    print("   • CSS:  css/style.css")
    print("   • JS:   js/app.js")
    print("\n🔄 Modo DEMO ativo - Dados simulados")
    print("\n💡 Endpoints disponíveis:")
    print("   GET  /api/sensors  - Dados dos sensores (requer auth)")
    print("   GET  /api/status   - Status do dispositivo (requer auth)")
    print("   POST /api/reset    - Reset de configuração (requer auth)")
    print("\n✨ Para parar o servidor: Ctrl+C")
    print("="*60 + "\n")
    
    app.run(
        host='0.0.0.0',  # Permite acesso de outros dispositivos na rede
        port=5000,
        debug=True,
        use_reloader=True
    )

File: api-standalone/dev-server.js
#!/usr/bin/env node
/**
 * Development Server - Local Mock API for Frontend Testing
 * Simula o comportamento da API do ESP32 rodando localmente
 */

const express = require('express');
const cors = require('cors');
const basicAuth = require('basic-auth');
const path = require('path');
const fs = require('fs');

const app = express();
const PORT = 5000;

// Arquivo para salvar credenciais
const CREDENTIALS_FILE = path.join(__dirname, 'credentials.json');

// Configuração de autenticação (valores padrão)
let API_USER = 'admin';
let API_PASS = 'admin123';
let WIFI_SSID = '';
let WIFI_PASSWORD = '';

// Carregar credenciais salvas (se existirem)
function loadCredentials() {
    try {
        if (fs.existsSync(CREDENTIALS_FILE)) {
            const data = fs.readFileSync(CREDENTIALS_FILE, 'utf8');
            const saved = JSON.parse(data);
            API_USER = saved.username || 'admin';
            API_PASS = saved.password || 'admin123';
            WIFI_SSID = saved.wifi_ssid || '';
            WIFI_PASSWORD = saved.wifi_password || '';
            console.log('✓ Credenciais carregadas do arquivo credentials.json');
            console.log(`   Usuário: ${API_USER}`);
            if (WIFI_SSID) {
                console.log(`   WiFi SSID: ${WIFI_SSID}`);
                console.log('   WiFi Password: ***');
            }
        } else {
            console.log('ℹ️ Usando credenciais padrão (arquivo não existe)');
        }
    } catch (error) {
        console.warn('⚠️ Erro ao carregar credenciais, usando valores padrão:', error.message);
    }
}

// Salvar credenciais em arquivo
function saveCredentials(username, password, wifiSSID, wifiPassword) {
    try {
        const data = {
            username: username,
            password: password,
            updated_at: new Date().toISOString()
        };
        
        // Adicionar WiFi apenas se fornecido
        if (wifiSSID) {
            data.wifi_ssid = wifiSSID;
            data.wifi_password = wifiPassword;
        } else if (WIFI_SSID) {
            // Manter WiFi existente se não foi fornecido novo
            data.wifi_ssid = WIFI_SSID;
            data.wifi_password = WIFI_PASSWORD;
        }
        
        fs.writeFileSync(CREDENTIALS_FILE, JSON.stringify(data, null, 2), 'utf8');
        console.log('✓ Credenciais salvas em credentials.json');
        return true;
    } catch (error) {
        console.error('❌ Erro ao salvar credenciais:', error.message);
        return false;
    }
}

// Carregar credenciais na inicialização
loadCredentials();

// Timestamp de início do servidor
const START_TIME = Date.now();

// Middleware
app.use(cors()); // Enable CORS for local development
app.use(express.json());

// Middleware de autenticação
function requireAuth(req, res, next) {
    const credentials = basicAuth(req);
    
    if (!credentials) {
        res.set('WWW-Authenticate', 'Basic realm="Login Required"');
        return res.status(401).json({ error: 'Authentication required' });
    }
    
    // Verificar username
    if (credentials.name !== API_USER) {
        res.set('WWW-Authenticate', 'Basic realm="Login Required"');
        return res.status(401).json({ error: 'Authentication required' });
    }
    
    // Tentar múltiplas formas de decodificar a senha (para suportar diferentes escapamentos do &)
    const receivedPass = credentials.pass;
    const validPasswords = [
        API_PASS,                                    // Original
        decodeURIComponent(API_PASS),               // URL decoded
        API_PASS.replace(/&/g, '%26'),              // URL encoded
        API_PASS.replace(/&/g, '&amp;'),            // HTML encoded
    ];
    
    // Também tentar decodificar a senha recebida
    const receivedVariants = [
        receivedPass,
        decodeURIComponent(receivedPass),
        receivedPass.replace(/%26/g, '&'),
        receivedPass.replace(/&amp;/g, '&'),
    ];
    
    let authenticated = false;
    for (const variant of receivedVariants) {
        if (validPasswords.includes(variant)) {
            authenticated = true;
            break;
        }
    }
    
    if (!authenticated) {
        res.set('WWW-Authenticate', 'Basic realm="Login Required"');
        return res.status(401).json({ error: 'Authentication required' });
    }
    
    next();
}

// Função para gerar dados simulados (mesma lógica do ESP32)
function generateDemoData() {
    const currentTime = Date.now() / 1000;
    const variance = Math.sin(currentTime / 30.0) * 0.1; // Variação lenta
    
    // Solar PV - simulando sol parcial
    const pvVoltage = 85.0 + variance * 10.0;
    const pvCurrent = 4.5 + variance * 2.0;
    const pvPower = pvVoltage * pvCurrent;
    
    // Carregador
    const chargerVoltage = 54.2 + variance * 0.5;
    const chargerCurrent = 8.2 + variance * 1.0;
    const chargerPower = 445.0 + variance * 50.0;
    
    // Bateria - simulando estado de carga
    const batteryVoltage = 52.8 + variance * 0.3;
    const batteryCurrent = 8.0 + variance * 1.5;
    const batteryPower = 422.0 + variance * 40.0;
    const batterySOC = 65.0 + variance * 5.0;
    const batteryTemp = 25.0 + variance * 2.0;
    
    // Inversor - simulando modo off-grid com carga
    const acVoltage = 220.0 + variance * 2.0;
    const acCurrent = 2.5 + variance * 0.5;
    const acFrequency = 50.0 + variance * 0.1;
    const acPower = 550.0 + variance * 50.0;
    const loadPercent = 27.5 + variance * 5.0;
    const dcVoltage = 52.8 + variance * 0.3;
    
    // Acumuladores - incrementando lentamente
    const timeHours = currentTime / 3600.0;
    const chargerAccumulated = 12.5 + (timeHours * 0.1) % 5.0;
    const totalCharged = 245.8 + (timeHours * 0.2) % 10.0;
    const totalDischarged = 198.3 + (timeHours * 0.15) % 8.0;
    
    const deviceTemp = 42.0 + variance * 3.0;
    
    return {
        charger: {
            voltage: Math.round(chargerVoltage * 10) / 10,
            current: Math.round(chargerCurrent * 100) / 100,
            power: Math.round(chargerPower),
            accumulated_power: Math.round(chargerAccumulated * 10) / 10
        },
        pv: {
            voltage: Math.round(pvVoltage * 10) / 10,
            current: Math.round(pvCurrent * 100) / 100,
            power: Math.round(pvPower)
        },
        battery: {
            voltage: Math.round(batteryVoltage * 10) / 10,
            current: Math.round(batteryCurrent * 100) / 100,
            power: Math.round(batteryPower),
            soc: Math.round(batterySOC),
            temperature: Math.round(batteryTemp * 10) / 10
        },
        inverter: {
            mode: "Off-Grid",
            mode_id: 3,
            ac_voltage: Math.round(acVoltage * 10) / 10,
            ac_current: Math.round(acCurrent * 100) / 100,
            ac_frequency: Math.round(acFrequency * 100) / 100,
            ac_power: Math.round(acPower),
            load_percent: Math.round(loadPercent),
            dc_voltage: Math.round(dcVoltage * 10) / 10,
            max_charge_current: 60,
            max_discharge_current: 60,
            accumulated_power: Math.round(totalDischarged * 10) / 10
        },
        totals: {
            total_charged: Math.round(totalCharged * 10) / 10,
            total_discharged: Math.round(totalDischarged * 10) / 10,
            device_temperature: Math.round(deviceTemp * 10) / 10
        },
        last_update: Math.floor(currentTime),
        uptime: Math.floor((Date.now() - START_TIME) / 1000),
        modbus_error: false,
        demo_mode: true
    };
}

// ==================== ROTAS DA API ====================

app.get('/api/sensors', requireAuth, (req, res) => {
    const data = generateDemoData();
    res.json(data);
});

app.get('/api/status', requireAuth, (req, res) => {
    res.json({
        device_name: "MUST Inverter API (Dev Server)",
        ip_address: "127.0.0.1",
        mac_address: "00:00:00:00:00:00",
        wifi_ssid: WIFI_SSID || "Development",
        wifi_rssi: -42,
        uptime_seconds: Math.floor((Date.now() - START_TIME) / 1000),
        free_heap: 200000,
        modbus_connected: false
    });
});

app.post('/api/reset', requireAuth, (req, res) => {
    res.json({ message: "Reset não disponível no dev server" });
});

app.get('/api/credentials', requireAuth, (req, res) => {
    // Retorna as credenciais atuais (sem a senha por segurança)
    res.json({
        username: API_USER,
        wifi_ssid: WIFI_SSID || null,
        has_wifi_password: WIFI_PASSWORD && WIFI_PASSWORD.length > 0
    });
});

app.get('/api/wifi/scan', requireAuth, (req, res) => {
    // Simular scan de redes WiFi (apenas 2.4 GHz - ESP32 não suporta 5 GHz)
    console.log('🔍 Simulando scan de redes WiFi 2.4 GHz...');
    
    const simulatedNetworks = [
        { ssid: 'CasteloAlto', rssi: -45, encryption: 'WPA2', channel: 6 },
        { ssid: 'NET_2G_123456', rssi: -55, encryption: 'WPA2', channel: 11 },
        { ssid: 'Vivo-Fibra-2.4G', rssi: -65, encryption: 'WPA2', channel: 1 },
        { ssid: 'Claro_WiFi', rssi: -72, encryption: 'WPA2', channel: 3 },
        { ssid: 'TIM-FIBRA-2.4G', rssi: -78, encryption: 'WPA2', channel: 9 },
        { ssid: 'WiFi-Guest', rssi: -85, encryption: 'Open', channel: 7 }
        // Nota: Redes 5 GHz são filtradas (ESP32 não suporta)
    ];
    
    // Ordenar por sinal (mais forte primeiro)
    simulatedNetworks.sort((a, b) => b.rssi - a.rssi);
    
    res.json({
        success: true,
        networks: simulatedNetworks,
        count: simulatedNetworks.length,
        note: 'Only 2.4 GHz networks (ESP32 compatible)'
    });
});

app.post('/api/credentials', requireAuth, (req, res) => {
    const { username, password, current_password, wifi_ssid, wifi_password } = req.body;
    
    // Validar senha atual
    if (current_password !== API_PASS) {
        return res.status(401).json({ 
            error: 'Current password is incorrect' 
        });
    }
    
    // Validar se há algo para alterar
    if (!username && !password && !wifi_ssid) {
        return res.status(400).json({ 
            error: 'No changes provided' 
        });
    }
    
    // Validar tamanho da senha
    if (password && password.length < 6) {
        return res.status(400).json({ 
            error: 'Password must be at least 6 characters' 
        });
    }
    
    // Validar WiFi
    if (wifi_ssid && !wifi_password) {
        return res.status(400).json({ 
            error: 'WiFi password is required when SSID is provided' 
        });
    }
    
    if (wifi_password && wifi_password.length < 8) {
        return res.status(400).json({ 
            error: 'WiFi password must be at least 8 characters' 
        });
    }
    
    // Avisar sobre limitações do caractere & na senha
    let warning = null;
    if (password && password.includes('&')) {
        warning = 'Atenção: O caractere "&" funciona via API mas pode não funcionar na caixa de login HTTP Basic Auth de alguns navegadores. Para máxima compatibilidade, evite usar "&" ou acesse via JavaScript/API.';
    }
    
    console.log('⚠️ Tentativa de alteração de credenciais:');
    console.log(`   Novo usuário: ${username || '(não alterado)'}`);
    console.log(`   Nova senha: ${password ? '***' : '(não alterada)'}`);
    if (wifi_ssid) {
        console.log(`   WiFi SSID: ${wifi_ssid}`);
        console.log(`   WiFi Password: ${wifi_password ? '***' : '(não alterada)'}`);
        console.log('   ⚠️ Nota: WiFi config é simulado no dev-server (apenas log)');
    }
    if (warning) {
        console.log(`   ⚠️ Aviso: ${warning}`);
    }
    
    // Definir valores finais
    const finalUsername = username || API_USER;
    const finalPassword = password || API_PASS;
    const finalWifiSSID = wifi_ssid || WIFI_SSID;
    const finalWifiPassword = wifi_password || WIFI_PASSWORD;
    
    // Salvar em arquivo
    if (saveCredentials(finalUsername, finalPassword, finalWifiSSID, finalWifiPassword)) {
        // Atualizar variáveis globais
        API_USER = finalUsername;
        API_PASS = finalPassword;
        
        if (wifi_ssid) {
            WIFI_SSID = finalWifiSSID;
            WIFI_PASSWORD = finalWifiPassword;
        }
        
        const response = { 
            success: true, 
            message: "Credentials updated successfully. Server will restart in 2 seconds.",
            username: finalUsername,
            note: "Dev-server will automatically restart to load new credentials"
        };
        
        if (warning) {
            response.warning = warning;
        }
        
        console.log('✅ Credenciais alteradas com sucesso!');
        console.log('🔄 Reiniciando servidor em 2 segundos...');
        
        // Enviar resposta e depois reiniciar o servidor
        res.json(response);
        
        // Aguardar 2 segundos e reiniciar o processo Node.js
        setTimeout(() => {
            console.log('\n🔄 Reiniciando dev-server...\n');
            process.exit(0); // Exit code 0 indica sucesso
        }, 2000);
    } else {
        res.status(500).json({
            error: "Failed to save credentials"
        });
    }
});

// ==================== SERVIR ARQUIVOS ESTÁTICOS ====================

// IMPORTANTE: Rotas específicas ANTES do express.static para ter controle de autenticação

// Rota raiz (requer autenticação)
app.get('/', requireAuth, (req, res) => {
    res.sendFile(path.join(__dirname, 'data', 'index.html'));
});

// Página de configuração (sem autenticação - modo AP)
app.get('/config.html', (req, res) => {
    res.sendFile(path.join(__dirname, 'data', 'config.html'));
});

// Página de teste de autenticação (sem autenticação - para debug)
app.get('/test-auth.html', (req, res) => {
    res.sendFile(path.join(__dirname, 'data', 'test-auth.html'));
});

// Endpoint para forçar logout (retorna 401 para limpar cache)
app.get('/logout', (req, res) => {
    res.set('WWW-Authenticate', 'Basic realm="Login Required"');
    res.status(401).send('Logout realizado. Feche o navegador ou use modo anônimo.');
});

// Página de configurações (requer autenticação)
app.get('/settings.html', requireAuth, (req, res) => {
    res.sendFile(path.join(__dirname, 'data', 'settings.html'));
});

// index.html também deve exigir autenticação
app.get('/index.html', requireAuth, (req, res) => {
    res.sendFile(path.join(__dirname, 'data', 'index.html'));
});

// Servir apenas CSS e JS sem autenticação (arquivos estáticos necessários)
app.use('/css', express.static(path.join(__dirname, 'data', 'css')));
app.use('/js', express.static(path.join(__dirname, 'data', 'js')));

// ==================== INICIALIZAÇÃO ====================

app.listen(PORT, '0.0.0.0', () => {
    console.log('\n' + '='.repeat(60));
    console.log('  🚀 MUST Inverter API - Development Server');
    console.log('='.repeat(60));
    console.log('\n📌 Servidor rodando em modo de desenvolvimento');
    console.log(`   URL Local:  http://localhost:${PORT}`);
    console.log(`   URL Rede:   http://0.0.0.0:${PORT}`);
    console.log('\n🔐 Credenciais de autenticação:');
    console.log(`   Usuário: ${API_USER}`);
    console.log(`   Senha: ${API_PASS}`);
    console.log('\n📁 Servindo arquivos de: ./data/');
    console.log('   • HTML: index.html, config.html');
    console.log('   • CSS:  css/style.css');
    console.log('   • JS:   js/app.js');
    console.log('\n🔄 Modo DEMO ativo - Dados simulados');
    console.log('\n💡 Endpoints disponíveis:');
    console.log('   GET  /api/sensors  - Dados dos sensores (requer auth)');
    console.log('   GET  /api/status   - Status do dispositivo (requer auth)');
    console.log('   POST /api/reset    - Reset de configuração (requer auth)');
    console.log('\n✨ Para parar o servidor: Ctrl+C');
    console.log('='.repeat(60) + '\n');
});

// Tratamento de erros
process.on('SIGINT', () => {
    console.log('\n\n👋 Servidor finalizado!');
    process.exit(0);
});
